module/VuFind/src/VuFind/Controller/HoldsController.php

```diff
@@ -102,4 +157,213 @@ public function listAction()
             ->collectRequestStats($view->recordList);
         return $view;
     }
+
+    /**
+     * Edit holds
+     *
+     * @return mixed
+     */
+    public function editAction()
+    {
+        // Stop now if the user does not have valid catalog credentials available:
+        if (!is_array($patron = $this->catalogLogin())) {
+            return $patron;
+        }
+
+        // Connect to the ILS:
+        $catalog = $this->getILS();
+
+        $holdConfig = $catalog->checkFunction('Holds', compact('patron'));
+        $selectedIds = $this->params()->fromPost('selectedIDS')
+            ?: $this->params()->fromQuery('selectedIDS');
+        if (empty($holdConfig['updateFields']) || empty($selectedIds)) {
+            // Shouldn't be here. Redirect to holds
+            return $this->redirect()->toRoute('holds-list');
+        }
+        // If the user input contains a value not found in the session
+        // legal list, something has been tampered with -- abort the process.
+        if (array_diff($selectedIds, $this->holds()->getValidIds())) {
+            $this->flashMessenger()
+                ->addErrorMessage('error_inconsistent_parameters');
+            return $this->redirect()->toRoute('holds-list');
+        }
+
+        $pickupLocations = $this->getPickupLocationsForEdit($patron, $selectedIds);
+
+        $gatheredDetails = $this->params()->fromPost('gatheredDetails', []);
+        if ($this->params()->fromPost('updateHolds')) {
+            if (!$this->csrf->isValid($this->params()->fromPost('csrf'))) {
+                throw new \VuFind\Exception\BadRequest(
+                    'error_inconsistent_parameters'
+                );
+            }
+
+            $updateFields = $this->getUpdateFieldsFromGatheredDetails(
+                $holdConfig,
+                $gatheredDetails,
+                $pickupLocations
+            );
+            if ($updateFields) {
+                $results
+                    = $catalog->updateHolds($selectedIds, $updateFields, $patron);
+                $successful = 0;
+                $failed = 0;
+                foreach ($results as $result) {
+                    if ($result['success']) {
```

Comment:
I think you could be clever with `$result['success'] ? ++$successful : ++failed;` here if you want to shave some lines of code. Feel free to disregard if you think that's too clever for our own good.